src/hexo/themes/documentation/layout/partials/docsearch.hbs

```diff
@@ -1,10 +1,28 @@
 <script src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js"></script>
 <script>
-    docsearch({
-        apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
-        indexName: 'sonarwhal',
-        inputSelector: '#search-input',
-        appId: '0R8E5SPAFP',
-        debug: false // set to true if need to debug css
-    });
+    (function () {
+        var search;
+        var previewSize = 500;
+        var transformInput = function () {
+            if (window.innerWidth > previewSize) {
+                // transform the input field to an algolia search input if there is enough space
+                search = docsearch({
+                    apiKey: '2ee58f908e6d720c4f72a9645ce0e857', // search-only API key, okay for frontend
+                    indexName: 'sonarwhal',
+                    inputSelector: '#search-input',
+                    appId: '0R8E5SPAFP',
+                    debug: false, // set to true if need to debug css
+                });
+            } else {
+                // transform the input field to a normal input field
+                // hitting enter will redirect to the search result page
+                if (search) {
+                    search.autocomplete.autocomplete.destroy();
+                }
+            }
+        };
+
+        window.addEventListener('resize', transformInput);
```

Comment:
@molant It looks like I can override the `handleselected` function. So I remove the the resize handler all together :).